modules/services/kanshi.nix

```diff
@@ -0,0 +1,201 @@
+          Modes are a combination of width and height (in pixels) and
+          a refresh rate (in Hz) that your display can be configured to use.
+          </literallayout>
+        '';
+      };
+
+      position = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "1600,0";
+        description = ''
+          <literallayout>
+          &lt;x&gt;,&lt;y&gt;
+
+          Places the output at the specified position in the global coordinates
+          space.
+          </literallayout>
+        '';
+      };
+
+      scale = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "2";
+        description = ''
+          Scales the output by the specified scale factor.
+        '';
+      };
+
+      transform = mkOption {
+        type = types.nullOr (types.enum [
+          "normal"
+          "90"
+          "180"
+          "270"
+          "flipped"
+          "flipped-90"
+          "flipped-180"
+          "flipped-270"
+        ]);
+        default = null;
+        description = ''
+          Sets the output transform.
+        '';
+      };
+    };
+  };
+
+  outputStr = { criteria, status, mode, position, scale, transform, ... }:
+    ''output "${criteria}"'' + optionalString (status != null) " ${status}"
+    + optionalString (mode != null) " mode ${mode}"
+    + optionalString (position != null) " position ${position}"
+    + optionalString (scale != null) " scale ${scale}"
+    + optionalString (transform != null) " transform ${transform}";
+
+  profileModule = types.submodule {
+    options = {
+
+      name = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Optional name of the profile.
+        '';
+      };
+
+      outputs = mkOption {
+        type = types.listOf outputModule;
+        default = [ ];
+        description = ''
+          Outputs configuration.
+        '';
+      };
+
+      exec = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Command executed after the profile is succesfully applied.
+        '';
+      };
+    };
+  };
+
+  profileStr = { name, outputs, exec, ... }:
+    ''
+      profile ${optionalString (name != null) "${name}"} {
+        ${concatStringsSep "\n  " (map outputStr outputs)}
+    '' + optionalString (exec != null) "  ${exec}" + ''
+      }
+    '';
+in {
+
+  meta.maintainers = [ maintainers.nurelin ];
+
+  options.services.kanshi = {
+    enable = mkEnableOption
+      "kanshi, a Wayland daemon that automatically configures outputs";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.kanshi;
+      defaultText = literalExample "pkgs.kanshi";
+      description = ''
+        kanshi derivation to use.
+      '';
+    };
+
+    profiles = mkOption {
+      type = types.listOf profileModule;
```

Comment:
Maybe it would be ok to make the profile name mandatory and turn this into an attribute set?